### ISSUE TITLE
using custom target concept 

### ISSUE THREAD

**Issue by MaxwellHogan** [jacobgil/pytorch-grad-cam#69]
I have used your code to experiment with grad-cam on YOLO, which I had to come up with a custom way of encoding the target - as opposed to a one-hot encoding of the target class 

I am also working on other architectures which have multiple outputs and have again had to come with a custom way of encoding the target

I have seen recently you have moved towards being able use the code as a library, I wondered if there were a means to use a custom target concept without having to modify the code so that it will make my task easier 

If not can I suggest it for future implementation

**Comment by jacobgil** [comment]
Hi,
Thanks for the suggestion.

Yes, refactoring to make it easier to work with object detection (for example) is something that I'm considering!

Currently, as you saw, it accepts only a target category and does:
1. forward pass
2. get the output score for that category 

For a more complex scenario (like object detection) , maybe would make sense for it to accept a function that gets the input tensor and target category and returns the loss, and let the user implement the custom logic there.

**Comment by jacobgil** [comment]
As a placeholder,
We have the 
set_loss function, which you can overridee with custom logic!
e.g

cam.set_loss = my_new_function.
